// @fillout/react/dist/index.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  FilloutFullScreenEmbed: () => FullScreen,
  FilloutPopupEmbed: () => Popup,
  FilloutPopupEmbedButton: () => PopupButton,
  FilloutSliderEmbed: () => Slider,
  FilloutSliderEmbedButton: () => SliderButton,
  FilloutStandardEmbed: () => Standard
});
module.exports = __toCommonJS(index_exports);

// #style-inject:#style-inject
function styleInject(css, { insertAt } = {}) {
  if (!css || typeof document === "undefined") return;
  const head = document.head || document.getElementsByTagName("head")[0];
  const style = document.createElement("style");
  style.type = "text/css";
  if (insertAt === "top") {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }
  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

// src/style.css
styleInject("@keyframes fillout-embed-loading {\n  0% {\n    transform: rotate(0deg);\n  }\n  100% {\n    transform: rotate(360deg);\n  }\n}\n.fillout-embed-popup-container {\n  padding: 40px 80px;\n}\n.fillout-embed-popup-iframe {\n  border-radius: 10px;\n  height: 100%;\n}\n.fillout-embed-popup-close {\n  position: absolute;\n  top: -15px;\n  right: -15px;\n  background: #171717;\n}\n.fillout-embed-popup-close:hover {\n  background: #272727;\n}\n.fillout-embed-slider-main {\n  position: relative;\n}\n.fillout-embed-slider-close {\n  padding: 20px 4px;\n}\n@media screen and (max-width: 480px) {\n  .fillout-embed-popup-container {\n    padding: 0;\n  }\n  .fillout-embed-popup-main {\n    display: flex;\n    flex-flow: column;\n    width: 100% !important;\n    height: 100% !important;\n  }\n  .fillout-embed-popup-iframe {\n    border-radius: 0;\n    height: unset;\n    flex-grow: 1;\n  }\n  .fillout-embed-popup-close {\n    position: unset;\n    top: 0;\n    right: 0;\n    margin: 4px 8px 4px auto;\n  }\n  .fillout-embed-slider-main {\n    flex-flow: column-reverse !important;\n    position: unset;\n    left: 0 !important;\n    right: 0 !important;\n  }\n  .fillout-embed-slider-iframe {\n    width: 100% !important;\n  }\n  .fillout-embed-slider-close {\n    padding: 6px;\n    border-radius: 50% !important;\n    margin: 4px 8px 4px auto;\n  }\n}\n");

// src/embeds/Standard.tsx
var import_react4 = __toESM(require("react"), 1);

// src/embed.ts
var import_react = require("react");
var FILLOUT_BASE_URL = "https://embed.fillout.com";
var generateEmbedId = () => {
  const min = 1e13;
  const max = 99999999999999;
  const randomNumber = Math.floor(Math.random() * (max - min + 1)) + min;
  return `${randomNumber}`;
};
var useFilloutEmbed = ({
  filloutId,
  domain,
  inheritParameters,
  parameters,
  dynamicResize
}) => {
  const [searchParams, setSearchParams] = (0, import_react.useState)();
  const [embedId, setEmbedId] = (0, import_react.useState)();
  (0, import_react.useEffect)(() => {
    setSearchParams(new URLSearchParams(window.location.search));
    setEmbedId(generateEmbedId());
  }, []);
  if (!searchParams || !embedId) return;
  const origin = domain ? `https://${domain}` : FILLOUT_BASE_URL;
  const iframeUrl = new URL(`${origin}/t/${encodeURIComponent(filloutId)}`);
  if (inheritParameters && searchParams) {
    for (const [key, value] of searchParams.entries()) {
      iframeUrl.searchParams.set(key, value);
    }
  }
  if (parameters) {
    for (const [key, value] of Object.entries(parameters)) {
      if (value) iframeUrl.searchParams.set(key, value);
    }
  }
  iframeUrl.searchParams.set("fillout-embed-id", embedId);
  if (dynamicResize) {
    iframeUrl.searchParams.set("fillout-embed-dynamic-resize", "true");
  }
  return {
    iframeUrl: iframeUrl.toString(),
    origin,
    embedId
  };
};

// src/components/Loading.tsx
var import_react2 = __toESM(require("react"), 1);
var Loading = () => /* @__PURE__ */ import_react2.default.createElement(
  "div",
  {
    style: {
      border: "solid 2px #aaa",
      borderBottomColor: "white",
      borderRadius: "50%",
      width: 32,
      height: 32,
      animation: "fillout-embed-loading 1s infinite linear"
    }
  }
);

// src/messages.ts
var import_react3 = require("react");
var useMessageListener = (embed, eventName, fn, options) => {
  const enabled = !(options == null ? void 0 : options.disabled);
  (0, import_react3.useEffect)(() => {
    if (embed && enabled) {
      const debug = location.href.includes("FILLOUT_EMBED_DEBUG");
      const listener = (event) => {
        try {
          if (event.origin === embed.origin && event.data.embedId === embed.embedId && event.data.type === eventName) {
            if (debug) {
              console.log(["fillout embed MESSAGE", eventName, event.data]);
            }
            fn(event.data);
          }
        } catch (err) {
        }
      };
      if (debug) console.log(["fillout embed MOUNT", eventName]);
      window.addEventListener("message", listener);
      return () => {
        if (debug) console.log(["fillout embed UNMOUNT", eventName]);
        window.removeEventListener("message", listener);
      };
    }
  }, [embed, eventName, fn, enabled]);
};

// src/events.ts
var useFilloutEvents = (embed, events) => {
  useMessageListener(
    embed,
    "form_init",
    (data) => {
      var _a;
      return (_a = events.onInit) == null ? void 0 : _a.call(events, data.submissionUuid);
    },
    { disabled: !events.onInit }
  );
  useMessageListener(
    embed,
    "page_change",
    (data) => {
      var _a;
      return (_a = events.onPageChange) == null ? void 0 : _a.call(events, data.submissionUuid, data.stepId);
    },
    { disabled: !events.onPageChange }
  );
  useMessageListener(
    embed,
    "form_submit",
    (data) => {
      var _a;
      return (_a = events.onSubmit) == null ? void 0 : _a.call(events, data.submissionUuid);
    },
    { disabled: !events.onSubmit }
  );
};

// src/embeds/Standard.tsx
var Standard = ({
  filloutId,
  domain,
  inheritParameters,
  parameters,
  dynamicResize,
  onInit,
  onPageChange,
  onSubmit
}) => {
  const [loading, setLoading] = (0, import_react4.useState)(true);
  const embed = useFilloutEmbed({
    filloutId,
    domain,
    inheritParameters,
    parameters,
    dynamicResize
  });
  useFilloutEvents(embed, { onInit, onPageChange, onSubmit });
  const [height, setHeight] = (0, import_react4.useState)();
  useMessageListener(
    embed,
    "form_resized",
    (data) => {
      const newHeight = data.size;
      if (typeof newHeight === "number") {
        setHeight(newHeight);
      }
    },
    { disabled: !dynamicResize }
  );
  return /* @__PURE__ */ import_react4.default.createElement(
    "div",
    {
      className: "fillout-standard-embed",
      style: {
        height: !dynamicResize ? "100%" : typeof height === "number" ? height : 256,
        transition: dynamicResize ? "height 150ms ease" : void 0
      }
    },
    loading && /* @__PURE__ */ import_react4.default.createElement(
      "div",
      {
        style: {
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          width: "100%",
          height: "100%",
          minHeight: 256
        }
      },
      /* @__PURE__ */ import_react4.default.createElement(Loading, null)
    ),
    embed && /* @__PURE__ */ import_react4.default.createElement(
      "iframe",
      {
        src: embed.iframeUrl,
        allow: "microphone; camera; geolocation",
        title: "Embedded Form",
        onLoad: () => setLoading(false),
        style: {
          width: !loading ? "100%" : 0,
          height: !loading ? "100%" : 0,
          opacity: !loading ? 1 : 0,
          borderRadius: 10,
          border: 0,
          minHeight: 256
        }
      }
    )
  );
};

// src/embeds/FullScreen.tsx
var import_react5 = __toESM(require("react"), 1);
var FullScreen = ({
  filloutId,
  domain,
  inheritParameters,
  parameters,
  onInit,
  onPageChange,
  onSubmit
}) => {
  const [loading, setLoading] = (0, import_react5.useState)(true);
  const embed = useFilloutEmbed({
    filloutId,
    domain,
    inheritParameters,
    parameters
  });
  useFilloutEvents(embed, { onInit, onPageChange, onSubmit });
  return /* @__PURE__ */ import_react5.default.createElement(
    "div",
    {
      style: {
        position: "fixed",
        top: 0,
        left: 0,
        width: "100%",
        height: "100%"
      }
    },
    loading && /* @__PURE__ */ import_react5.default.createElement(
      "div",
      {
        style: {
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          width: "100%",
          height: "100%"
        }
      },
      /* @__PURE__ */ import_react5.default.createElement(Loading, null)
    ),
    embed && /* @__PURE__ */ import_react5.default.createElement(
      "iframe",
      {
        src: embed.iframeUrl,
        allow: "microphone; camera; geolocation",
        title: "Embedded Form",
        style: {
          border: 0,
          width: !loading ? "100%" : 0,
          height: !loading ? "100%" : 0,
          opacity: !loading ? 1 : 0
        },
        onLoad: () => setLoading(false)
      }
    )
  );
};

// src/embeds/Popup.tsx
var import_react7 = __toESM(require("react"), 1);

// src/components/Portal.tsx
var import_react6 = __toESM(require("react"), 1);
var import_react_dom = require("react-dom");
var Portal = ({ children }) => {
  const [render, setRender] = (0, import_react6.useState)(false);
  (0, import_react6.useEffect)(() => {
    setRender(true);
  }, []);
  return render ? (0, import_react_dom.createPortal)(children, document.body) : /* @__PURE__ */ import_react6.default.createElement(import_react6.default.Fragment, null);
};

// src/embeds/Popup.tsx
var Popup = (_a) => {
  var _b = _a, {
    isOpen: _isOpen,
    onClose: _onClose,
    width,
    height
  } = _b, props = __objRest(_b, [
    "isOpen",
    "onClose",
    "width",
    "height"
  ]);
  const [isOpen, setIsOpen] = (0, import_react7.useState)(false);
  (0, import_react7.useEffect)(() => {
    if (_isOpen) setTimeout(() => setIsOpen(true), 10);
    else setIsOpen(false);
  }, [_isOpen]);
  const onClose = () => {
    if (!isOpen) return;
    setIsOpen(false);
    setTimeout(_onClose, 250);
  };
  return /* @__PURE__ */ import_react7.default.createElement(Portal, null, /* @__PURE__ */ import_react7.default.createElement(
    PopupContainer,
    {
      isOpen: _isOpen,
      isOpenAnimate: isOpen,
      onClose,
      width,
      height
    },
    _isOpen && /* @__PURE__ */ import_react7.default.createElement(PopupContent, __spreadValues({ onClose }, props))
  ));
};
var PopupContent = ({
  filloutId,
  domain,
  inheritParameters,
  parameters,
  onClose,
  onInit,
  onPageChange,
  onSubmit
}) => {
  const [loading, setLoading] = (0, import_react7.useState)(true);
  const embed = useFilloutEmbed({
    filloutId,
    domain,
    inheritParameters,
    parameters
  });
  useFilloutEvents(embed, { onInit, onPageChange, onSubmit });
  return /* @__PURE__ */ import_react7.default.createElement(import_react7.default.Fragment, null, !loading && /* @__PURE__ */ import_react7.default.createElement(CloseButton, { onClick: onClose }), embed && /* @__PURE__ */ import_react7.default.createElement(
    "iframe",
    {
      src: embed.iframeUrl,
      allow: "microphone; camera; geolocation",
      title: "Embedded Form",
      className: "fillout-embed-popup-iframe",
      style: {
        border: 0,
        width: "100%",
        opacity: !loading ? 1 : 0
      },
      onLoad: () => setLoading(false)
    }
  ), loading && /* @__PURE__ */ import_react7.default.createElement(
    "div",
    {
      style: {
        position: "fixed",
        top: 0,
        left: 0,
        width: "100%",
        height: "100%",
        display: "flex",
        justifyContent: "center",
        alignItems: "center"
      }
    },
    /* @__PURE__ */ import_react7.default.createElement(Loading, null)
  ));
};
var PopupContainer = ({
  children,
  isOpen,
  isOpenAnimate,
  onClose,
  width,
  height
}) => /* @__PURE__ */ import_react7.default.createElement(
  "div",
  {
    onClick: onClose,
    className: "fillout-embed-popup-container",
    style: {
      position: "fixed",
      top: 0,
      left: 0,
      width: "100%",
      height: "100%",
      background: "rgba(0, 0, 0, 0.65)",
      backdropFilter: "blur(4px)",
      transition: "opacity 0.25s ease-in-out",
      zIndex: 1e13,
      boxSizing: "border-box",
      opacity: isOpenAnimate ? 1 : 0,
      display: isOpen ? "flex" : "none",
      justifyContent: "center",
      alignItems: "center"
    }
  },
  /* @__PURE__ */ import_react7.default.createElement(
    "div",
    {
      className: "fillout-embed-popup-main",
      style: {
        position: "relative",
        width: width || "100%",
        height: height || "100%",
        maxWidth: "100%",
        maxHeight: "100%"
      }
    },
    children
  )
);
var CloseButton = ({ onClick }) => /* @__PURE__ */ import_react7.default.createElement(
  "button",
  {
    onClick: (e) => {
      e.stopPropagation();
      onClick();
    },
    className: "fillout-embed-popup-close",
    style: {
      width: 24,
      height: 24,
      textAlign: "center",
      cursor: "pointer",
      transition: "opacity 0.5s ease-in-out",
      textDecoration: "none",
      color: "white",
      borderRadius: "50%",
      padding: 6,
      boxSizing: "content-box",
      border: 0
    }
  },
  /* @__PURE__ */ import_react7.default.createElement(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      fill: "none",
      viewBox: "0 0 24 24",
      stroke: "currentColor",
      strokeWidth: "2"
    },
    /* @__PURE__ */ import_react7.default.createElement(
      "path",
      {
        strokeLinecap: "round",
        strokeLinejoin: "round",
        d: "M6 18L18 6M6 6l12 12"
      }
    )
  )
);

// src/embeds/PopupButton.tsx
var import_react9 = __toESM(require("react"), 1);

// src/components/Button.tsx
var import_react8 = __toESM(require("react"), 1);
var buttonSizeStyles = {
  small: {
    padding: "8px 12px",
    fontSize: 16,
    borderRadius: 28
  },
  medium: {
    padding: "10px 14px",
    fontSize: 18,
    borderRadius: 32
  },
  large: {
    padding: "12px 14px",
    fontSize: 20,
    borderRadius: 32
  }
};
var buttonFloatStyles = {
  bottomRight: {
    position: "fixed",
    bottom: 32,
    right: 32,
    zIndex: 9999999
  },
  bottomLeft: {
    position: "fixed",
    bottom: 32,
    left: 32,
    zIndex: 9999999
  }
};
var hexToRgb = (hex) => {
  if (typeof hex !== "string" || !/^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/.test(hex)) {
    return [59, 130, 246];
  }
  let bigint = parseInt(hex.slice(1), 16);
  let r = bigint >> 16 & 255;
  let g = bigint >> 8 & 255;
  let b = bigint & 255;
  return [r, g, b];
};
var getLuminance = (hexColor) => {
  let [r, g, b] = hexToRgb(hexColor);
  const getComponent = (color) => {
    color /= 255;
    return color <= 0.03928 ? color / 12.92 : Math.pow((color + 0.055) / 1.055, 2.4);
  };
  r = getComponent(r);
  g = getComponent(g);
  b = getComponent(b);
  return 0.2126 * r + 0.7152 * g + 0.0722 * b;
};
var Button = ({
  onClick,
  text = "Open form",
  color = "#3b82f6",
  size = "medium",
  float
}) => {
  const textColor = getLuminance(color) > 0.5 ? "black" : "white";
  return /* @__PURE__ */ import_react8.default.createElement(
    "button",
    {
      onClick,
      style: __spreadValues(__spreadValues({
        cursor: "pointer",
        fontFamily: "Helvetica, Arial, sans-serif",
        display: "inline-block",
        maxWidth: "100%",
        overflow: "hidden",
        textOverflow: "ellipsis",
        textDecoration: "none",
        fontWeight: "bold",
        textAlign: "center",
        margin: 0,
        border: "medium",
        boxShadow: "0px 2px 5px rgba(0, 0, 0, 0.2)",
        backgroundColor: color,
        color: textColor || "white"
      }, buttonSizeStyles[size]), float && buttonFloatStyles[float])
    },
    text
  );
};

// src/embeds/PopupButton.tsx
var PopupButton = ({
  filloutId,
  domain,
  inheritParameters,
  parameters,
  width,
  height,
  onInit,
  onPageChange,
  onSubmit,
  text,
  color,
  size,
  float
}) => {
  const [isOpen, setIsOpen] = (0, import_react9.useState)(false);
  return /* @__PURE__ */ import_react9.default.createElement(import_react9.default.Fragment, null, /* @__PURE__ */ import_react9.default.createElement(
    Button,
    {
      onClick: () => setIsOpen(true),
      text,
      color,
      size,
      float
    }
  ), /* @__PURE__ */ import_react9.default.createElement(
    Popup,
    {
      filloutId,
      domain,
      inheritParameters,
      parameters,
      isOpen,
      onClose: () => setIsOpen(false),
      onInit,
      onPageChange,
      onSubmit,
      width,
      height
    }
  ));
};

// src/embeds/Slider.tsx
var import_react10 = __toESM(require("react"), 1);
var Slider = (_a) => {
  var _b = _a, {
    isOpen: _isOpen,
    onClose: _onClose
  } = _b, props = __objRest(_b, [
    "isOpen",
    "onClose"
  ]);
  const [isOpen, setIsOpen] = (0, import_react10.useState)(false);
  (0, import_react10.useEffect)(() => {
    if (_isOpen) setTimeout(() => setIsOpen(true), 10);
    else setIsOpen(false);
  }, [_isOpen]);
  const onClose = () => {
    if (!isOpen) return;
    setIsOpen(false);
    setTimeout(_onClose, 250);
  };
  return /* @__PURE__ */ import_react10.default.createElement(Portal, null, /* @__PURE__ */ import_react10.default.createElement(
    SliderContainer,
    {
      isOpen: _isOpen,
      isOpenAnimate: isOpen,
      onClose
    },
    _isOpen && /* @__PURE__ */ import_react10.default.createElement(SliderContent, __spreadValues({ onClose }, props))
  ));
};
var SliderContent = ({
  filloutId,
  domain,
  inheritParameters,
  parameters,
  sliderDirection = "right",
  onClose,
  onInit,
  onPageChange,
  onSubmit
}) => {
  const [loading, setLoading] = (0, import_react10.useState)(true);
  const embed = useFilloutEmbed({
    filloutId,
    domain,
    inheritParameters,
    parameters
  });
  useFilloutEvents(embed, { onInit, onPageChange, onSubmit });
  const sliderLeft = sliderDirection === "left";
  const sliderOpen = !loading;
  return /* @__PURE__ */ import_react10.default.createElement(import_react10.default.Fragment, null, loading && /* @__PURE__ */ import_react10.default.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        width: "100%",
        height: "100%"
      }
    },
    /* @__PURE__ */ import_react10.default.createElement(Loading, null)
  ), /* @__PURE__ */ import_react10.default.createElement(
    "div",
    {
      className: "fillout-embed-slider-main",
      style: __spreadValues({
        display: "flex",
        justifyContent: sliderLeft ? "start" : "end",
        flexDirection: sliderLeft ? "row" : "row-reverse",
        alignItems: "center",
        height: !loading ? "100%" : 0,
        transitionProperty: sliderLeft ? "left" : "right",
        transitionDuration: "0.25s",
        transitionTimingFunction: "ease-in-out"
      }, sliderLeft ? {
        left: sliderOpen ? 0 : "-80%"
      } : {
        right: sliderOpen ? 0 : "-80%"
      })
    },
    embed && /* @__PURE__ */ import_react10.default.createElement(
      "iframe",
      {
        src: embed.iframeUrl,
        allow: "microphone; camera; geolocation",
        title: "Embedded Form",
        className: "fillout-embed-slider-iframe",
        style: {
          border: 0,
          width: !loading ? "80%" : 0,
          height: !loading ? "100%" : 0,
          opacity: !loading ? 1 : 0
        },
        onLoad: () => setLoading(false)
      }
    ),
    !loading && /* @__PURE__ */ import_react10.default.createElement(CloseButton2, { onClick: onClose, sliderLeft })
  ));
};
var SliderContainer = ({
  children,
  isOpen,
  isOpenAnimate,
  onClose
}) => /* @__PURE__ */ import_react10.default.createElement(
  "div",
  {
    onClick: onClose,
    style: {
      position: "fixed",
      top: 0,
      left: 0,
      width: "100%",
      height: "100%",
      background: "rgba(0, 0, 0, 0.65)",
      transition: "opacity 0.25s ease-in-out",
      zIndex: 1e13,
      opacity: isOpenAnimate ? 1 : 0,
      display: isOpen ? "block" : "none"
    }
  },
  children
);
var CloseButton2 = ({
  onClick,
  sliderLeft
}) => /* @__PURE__ */ import_react10.default.createElement(
  "button",
  {
    onClick: (e) => {
      e.stopPropagation();
      onClick();
    },
    className: "fillout-embed-slider-close",
    style: __spreadValues({
      border: 0,
      display: "flex",
      background: "#171717",
      color: "white",
      cursor: "pointer"
    }, sliderLeft ? {
      borderTopRightRadius: 15,
      borderBottomRightRadius: 15
    } : {
      borderTopLeftRadius: 15,
      borderBottomLeftRadius: 15
    })
  },
  /* @__PURE__ */ import_react10.default.createElement(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      fill: "none",
      viewBox: "0 0 24 24",
      stroke: "currentColor",
      strokeWidth: "2",
      style: {
        width: 24,
        height: 24
      }
    },
    /* @__PURE__ */ import_react10.default.createElement(
      "path",
      {
        strokeLinecap: "round",
        strokeLinejoin: "round",
        d: "M6 18L18 6M6 6l12 12"
      }
    )
  )
);

// src/embeds/SliderButton.tsx
var import_react11 = __toESM(require("react"), 1);
var SliderButton = ({
  filloutId,
  domain,
  inheritParameters,
  parameters,
  sliderDirection,
  onInit,
  onPageChange,
  onSubmit,
  text,
  color,
  size,
  float
}) => {
  const [isOpen, setIsOpen] = (0, import_react11.useState)(false);
  return /* @__PURE__ */ import_react11.default.createElement(import_react11.default.Fragment, null, /* @__PURE__ */ import_react11.default.createElement(
    Button,
    {
      onClick: () => setIsOpen(true),
      text,
      color,
      size,
      float
    }
  ), /* @__PURE__ */ import_react11.default.createElement(
    Slider,
    {
      filloutId,
      domain,
      inheritParameters,
      parameters,
      sliderDirection,
      isOpen,
      onClose: () => setIsOpen(false),
      onInit,
      onPageChange,
      onSubmit
    }
  ));
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  FilloutFullScreenEmbed,
  FilloutPopupEmbed,
  FilloutPopupEmbedButton,
  FilloutSliderEmbed,
  FilloutSliderEmbedButton,
  FilloutStandardEmbed
});
